### ISSUE TITLE
Preset names lost when applying style

### ISSUE THREAD

**Issue by kofa73** [darktable-org/darktable#15805]
### Describe the bug

I have a preset that contains 2 instances of _diffuse or sharpen_, both built-in ones:
- sharpen demosaicing: AA filter
- local contrast

Previously, when I applied the style, the two instances were shown accordingly. Now, they are shown as:
![image](https://github.com/darktable-org/darktable/assets/3177739/921466a0-fe76-4c88-b4ad-62d59c6a4a0d)

The style contains:
```xml
        <plugin>
            <num>17</num>
            <module>2</module>
            <operation>diffuse</operation>
            <op_params>
                0100000000000000080000000000803f000000000000803f0000803f0000803f0000803f00000000000080be000080be000080be000080be00000000
            </op_params>
            <enabled>1</enabled>
            <blendop_params>gz09eJxjYGBgYAFiCQYYOOHEgAZY0QVwggZ7CB6pfOygYtaVAyCMi48L/AcCEA0AmawnoA==</blendop_params>
            <blendop_version>12</blendop_version>
            <multi_priority>1</multi_priority>
            <multi_name>sharpen demosaicing: AA filter</multi_name>
            <multi_name_hand_edited>0</multi_name_hand_edited>
        </plugin>
        <plugin>
            <num>16</num>
            <module>2</module>
            <operation>diffuse</operation>
            <op_params>
                0a00000000000000800100000000803f0000803f000020c00000000000000000000020c000000000000000bf0000000000000000000000bf00020000
            </op_params>
            <enabled>1</enabled>
            <blendop_params>gz09eJxjYGBgYAFiCQYYOOHEgAZY0QVwggZ7CB6pfOygYtaVAyCMi48L/AcCEA0AmawnoA==</blendop_params>
            <blendop_version>12</blendop_version>
            <multi_priority>0</multi_priority>
            <multi_name>local contrast</multi_name>
            <multi_name_hand_edited>0</multi_name_hand_edited>
        </plugin>
```

### Steps to reproduce

Apply the attached style to a photo.

### Expected behavior

Display the module instance names

### Where did you obtain darktable from?

self compiled

### darktable version

4.5.0+1391~g0a39901aa5-dirty (because of the tone-eq experiment from https://github.com/darktable-org/darktable/issues/15743#issuecomment-1828292364)

### What OS are you using?

Linux

### What is the version of your OS?

Ubuntu 23.10

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

not relevant

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

[000-sigmoid standard.dtstyle.txt](https://github.com/darktable-org/darktable/files/13533768/000-sigmoid.standard.dtstyle.txt)

**Comment by TurboGit** [comment]
Most probably because the name is not hand-edited (0 in `<multi_name_hand_edited>`). Then one applied the name is computed based on the current params & blend-params. Since the later has been changed recently there is no match and then the name get created based on the standard naming.

**Comment by kofa73** [comment]
Indeed, `blendop_version` changed from 12 to 13, and `blendop_params` also changed, but there is actually no mask of any kind applied, so this is rather annoying. `op_params` did not change.

**Comment by TurboGit** [comment]
Right, confusing but that's the way presets are matched (i.e. checking iop_params and blend_params). The only way to get stable naming is to hand-edit the name, so in your case create your own presets an create the style based on it. This was discussed and we don't want to be using the preset name if hand-edited is not set, it was too disturbing for some people.

**Comment by kofa73** [comment]
The style includes the preset name. So if there is no match by iop_params and blend_params, maybe fall back to the stored name, even if multi_name_hand_edited is not set?

Otherwise, there is no point in storing multi_name in the style, unless multi_name_hand_edited = 1 (since it will never be used). If multi_name were only stored when hand-edited, multi_name_hand_edited could be removed, as multi_name != null would imply hand_edited.

**Comment by kofa73** [comment]
I'll try to summarise the current situation, which is very complicated and, I think, contradictory. I'll get back later, probably not today.

**Comment by TurboGit** [comment]
Well @elstoc let's be honest, part of the complexity has been added because the initial behavior was not ok to you.

Anyway, @kofa73 thanks for this summary. I'll read it. If we agree on the current situation we'll need to put a proper solution on the table, for all of us.

**Comment by elstoc** [comment]
> part of the complexity has been added because the initial behavior was not ok to you

If you say so. I did offer a few suggestions to make it less confusing, but IMO it was just getting worse as the discussion continued, so I gave up in the knowledge that I could switch it off. I still haven't documented it in the manual yet though because it's hard to describe beyond a simple sentence "the module label might get automatically updated to something unless you disable the preference"

**Comment by TurboGit** [comment]
Let's start step-by-step.

> If neither the module, nor the module instance from which the preset was made have a manually edited name, the name of the preset will be used as the (default) name of the module instance.

This is good, right?

> If the module has no manually set name, but the module instance on which the preset was based had one, the manually set name of the original module instance, stored inside the preset, is used as the name of the module instance; the name will be marked as manually set on the module instance.

This is also good, right?

**Comment by TurboGit** [comment]
I'd prefer to concentrate first on the problem than proposing solutions. The problem may be clear to you but not to me. That's why I'm trying to get OK or NOT OK on the statement listed by @kofa73. When we'll have a clear idea of what is wrong or annoying then we will try to find a solution. Having new ticks and field is maybe not the best solution GUI wise.

**Comment by elstoc** [comment]
The problem for me is that there's nowhere on the UI that I can find out what will happen when I add a preset, and nowhere that I can set/change it on a per-preset basis. That makes it hard if not impossible to document what will happen as it may well depend on something I did inadvertently (changed the label) when creating the preset. The only way to fix that is to drop the preset and create a new one (or manually edit the db).

I guess this issue impacts all three points

1. I'm happy that applying a preset to an instance that has already been manually renamed, will never alter the name. I don't like "[if the] module instance on which the preset was based had [a manually created name], the manually set name of the original module instance [is set to any instance on which the preset is subsequently applied]" because this is dependent on a hidden field that I can neither see nor update, and I might not have intended that behaviour when I created the preset. Having to drop and recreate a preset to fix this is a pain.

2. Same issues as point 1. Again I don't have an issue with auto-setting the module name based on the instance against which it was created but I'd rather see this in the UI and be able to set/update it manually.

3. If I'm understanding this point I think I'm fine with this -- any manually edited instance never has its label updated if it's been manually set

The reason I jump to the solution is that it's easier to get my head round than trying to describe the issues with the current solution. I hope this has clarified things.

From an implementation point-of-view, I'd say that the cleanest thing is to only store manually edited names against the module instance, otherwise (if the manual name is empty) look up the name that's stored against the preset (and in my suggested implementation that will only ever come from a single user-exposed field, that may or may not be the same as the preset name).

**Comment by elstoc** [comment]
If we're really looking for a clean UI I can't really see much point to the "only show this preset" tickbox or the "description" field so we could remove them?

**Comment by kofa73** [comment]
I think hand-edited names should not be saved to presets:
- it leads to confusion ('why did applying this preset assign a name different from the name of the presets?' -- the end-user has no way to 'look inside the preset' to figure out the instance name stored inside);
- sometimes I assign module instance names in the context of the current image, but find that the settings could be useful for many images, so I create a preset, and assign it a generally meaningful name. However, currently the hand edited name, based on the image context is stored and then restored when the preset is applied to other images.

**Comment by kofa73** [comment]
Sorry about the slow response.

I consider 1 as 'good', 2 as 'bad' (see my post above regarding contexts), 3 as 'bad' (but I see your point, so I'm no longer so convinced).

I think there's no universal solution, and thorough documentation with an explanation (like yours for 3).

**Comment by EC1000** [comment]
For me, point 2 is good: it is particularly useful for auto-apply presets (that is the only way to define a manually set name that will stay consistent between images, to later copy-paste parameters easily), see initial discussion in https://github.com/darktable-org/darktable/issues/13982.

I agree that the way to edit the label that will be save is: 
a) not obvious to find (logical, but not obvious),
b) somehow cumbersome if you want to save a preset without it (remove label, save preset, rewrite label) or edit a preset to remove it (apply preset, modify label, save preset).
I am ok with the way it works, since I know how to proceed and I understand the motivation for not adding a new field, but I also understand it can generate misunderstandings.
